### PR TITLE
Implement IntoIterator for Receiver

### PR DESCRIPTION
**Pull request by rapha**

### CODE REVIEW

**Comment by alexcrichton** [comment]
This seems like a good idea to me, so thanks @rapha! This does, however, probably force our hand in having this new functionality be insta-`#[stable]`. I'm personally OK with that, however.

Could you change the imports of `core::iter::IntoIterator` to just `iter::IntoIterator` as well?

**Comment by rapha** [comment]
Thanks. I added doc and stability annotations to match those on the Iter type, adding that it was an "owning" iterator.

**Comment by rapha** [comment]
Do you need me to do anything more on this?

**Comment by alexcrichton** [comment]
Ah thanks for the reminder @rapha! I intended on just running this by some others to get their opinion as well. I believe now this is good to go! In the meantime, however, we've bumped the version number, so could you update the `since` tag here to `1.1.0` instead of `1.0.0`? After that I'll r+

**Comment by bluss** [comment]
Shouldn't it have another IntoIteraor impl to match the `fn iter(&self)` method too?

**Comment by alexcrichton** [comment]
@bluss in theory, yes, but it's not necessarily required one way or another as it's backwards compatible to add. @rapha would you be interested in adding this, however?

@rapha ah I forgot this part, but you'll also need to choose a unique feature name for this as the same feature name can't have multiple versions it became stable in (e.g. the `rust1` feature here was stable as of `1.0.0`, not `1.1.0`

**Comment by rapha** [comment]
@bluss Is this the implementation of IntoIterator for &Receiver what you had in mind?

File: src/libstd/sync/mpsc/mod.rs [hunk 1]
```diff
@@ -306,6 +306,14 @@ pub struct Iter<'a, T: 'a> {
     rx: &'a Receiver<T>
 }
 
+/// An owning iterator over messages on a receiver, this iterator will block
+/// whenever `next` is called, waiting for a new message, and `None` will be
+/// returned when the corresponding channel has hung up.
+#[stable(feature = "rust1_1", since = "1.1.0")]
```
Comment (alexcrichton) [inline src/libstd/sync/mpsc/mod.rs]: Ah sorry by new feature name I meant more something along the lines of `receiver_into_iter` or something like that

**Comment by alexcrichton** [comment]
@rapha looks good to me! Could you also squash the commits together?

**Comment by rapha** [comment]
Done

File: src/libstd/sync/mpsc/mod.rs [hunk 2]
```diff
@@ -899,6 +907,29 @@ impl<'a, T> Iterator for Iter<'a, T> {
     fn next(&mut self) -> Option<T> { self.rx.recv().ok() }
 }
 
+#[stable(feature = "rust1_1", since = "1.1.0")]
```
Comment (alexcrichton) [inline src/libstd/sync/mpsc/mod.rs]: One last instance of `rust1_1`

**Comment by bluss** [comment]
Yes that's great. It's consistent now that the .iter() method has a corresponding IntoIterator impl, thank you!

**Pull request merged**